static/src/javascripts/projects/facia/modules/ui/live-blog-updates.js

```diff
@@ -0,0 +1,228 @@
+            if (vPosition > 0 && vPosition < viewportHeightPx) {
+                setTimeout(() => {
+                    const $el = bonzo(el);
+
+                    fastdomPromise.write(() => {
+                        $el.removeClass(
+                            'fc-item__liveblog-blocks__inner--offset'
+                        );
+                    });
+                }, immediate ? 0 : animateDelayMs);
+                return true;
+            }
+            return false;
+        });
+
+const animateBlocks = (el: Element): void => {
+    maybeAnimateBlocks(el).then(didAnimate => {
+        let animateOnScroll;
+
+        if (!didAnimate) {
+            animateOnScroll = debounce(() => {
+                maybeAnimateBlocks(el, true).then(animated => {
+                    if (animated) {
+                        mediator.off('window:throttledScroll', animateOnScroll);
+                    }
+                });
+            }, animateAfterScrollDelayMs);
+
+            mediator.on('window:throttledScroll', animateOnScroll);
+        }
+    });
+};
+
+const showBlocks = (
+    articleId: string,
+    targets: Array<Element>,
+    blocks: Array<Block>,
+    oldBlockDate: number
+): void => {
+    const fakeUpdate = isUndefined(oldBlockDate);
+
+    targets.forEach(element => {
+        let hasNewBlock = false;
+
+        const wrapperClasses = [
+            'fc-item__liveblog-blocks__inner',
+            'u-faux-block-link__promote',
+        ];
+
+        const blocksHtml = blocks
+            .slice(0, 2)
+            .map((block, index) => {
+                if (
+                    !hasNewBlock &&
+                    (block.publishedDateTime > oldBlockDate || fakeUpdate)
+                ) {
+                    block.isNew = true;
+                    hasNewBlock = true;
+                    wrapperClasses.push(
+                        'fc-item__liveblog-blocks__inner--offset'
+                    );
+                }
+                return renderBlock(articleId, block, index);
+            })
+            .slice(0, hasNewBlock ? 2 : 1);
+
+        const el = bonzo.create(
+            `<div class="${wrapperClasses.join(' ')}">${blocksHtml.join(
+                ''
+            )}</div>`
+        );
+
+        const $element = bonzo(element);
+
+        fastdomPromise
+            .write(() => {
+                $element.empty().append(el);
+            })
+            .then(() => {
+                if (hasNewBlock) {
+                    animateBlocks(el[0]);
+                }
+            });
+    });
+};
+
+const sanitizeBlocks = (blocks: Array<Block>): Array<Block> =>
+    blocks.filter(
+        block =>
+            block.id &&
+            block.publishedDateTime &&
+            block.body &&
+            block.body.length >= 10
+    );
+
+const showUpdatesFromLiveBlog = (): Promise<void> =>
+    fastdomPromise
+        .read(() => {
+            const elementsById: Map<string, Array<Element>> = new Map();
```

Comment:
- `forEach` on a `Map` iterates over the key and value, as opposed to an Object literal where you can get the keys. This allowed for a more direct translation from the underscore equivalent.
- Defining it as a `Map` also allows you to be explicit that this homogenous structure where all the keys and all the values are of the same type. 
- I'd use an object literal where the keys were statically known, but it seems more natural to use a `Map` where the keys are dynamically assigned.